source/client-side-encryption/client-side-encryption.rst

```diff
@@ -545,6 +599,11 @@ When such a state is detected, libmongocrypt_ will call back to the driver by
 entering the ``MONGOCRYPT_CTX_NEED_KMS_CREDENTIALS`` state, upon which the
 driver should fill in the KMS options automatically.
 
+Automatic credentials are only supported for the KMS provider types ``aws``,
+``gcp``, and ``azure``. KMS providers containing a name (e.g. ``aws:myname``) do
+not support automatic credentials. Attempting to configure a KMS provider with a
```

Comment:
> Named KMS providers do not support "Automatic Credentials".

It would be very simple to add support for this in Python, why do we have this limitation?

Reply:
See [Q1: Should on-demand KMS credentials be supported for named KMS providers?](https://docs.google.com/document/d/1IUxkgGFx5VLjMr7HWxBOXerN0pUZwcqWb11gLoRtcJs/edit#heading=h.tdbcb3b33s9z).

Reply:
> On-demand credentials are primarily intended to support passing credentials assigned in an environment. Supporting on-demand credentials for more than one KMS provider of the same type is not expected to be useful. Supporting on-demand KMS credentials would require added work in drivers inspecting the KMS providers ([example in Java](https://github.com/mongodb/mongo-java-driver/blob/616ab9d32470d5f9aec3abdd3c3664256c49f52f/driver-core/src/main/com/mongodb/internal/capi/MongoCryptHelper.java#L114-L131)) when 
obtaining credentials.

Seems fine to me since we can always add support later if needed. Could we include this rationale in the spec?

Reply:
Added rationale.